## 문제
- 주어진 데이터(data/toothgrowth.csv)에는 기니피그 60마리의 치아 길이, 투여한 비타민 C 종류, 투여량이 저장되어 있다.  오렌지주스를 투여받은 기니피그의 치아 길이 평균이 아스코르브산을 투여받은 기니피그의 치아 길이 평균과 일치하는지 t-검정(Two Sample t-test)를 통해 답하고자 한다. 가설은 아래와 같다.
- 가설검정
    + 귀무가설 : H0: mu.X - mu.Y = 0
    + 대립가설 : H1: mu.X - mu.Y != 0
- 데이터의 변수
    + len : 치아의 길이
    + supp : 투여한 약제(VC: 아스코르브산, OJ: 오렌지주스)
    + does : 투여량
- 단, 데이터의 각 변수들은 정규분포를 만족하며, 두 그룹은 등분산을 띤다.
- 문제 1: 표본평균 $\overline{X_{OJ}}$, $\overline{X_{VC}}$를 구하시오.
- 문제 2: 위의 가설을 검정하기 위한 검정통계량을 구하시오. (반올림하여 소숫점 둘째 자리까지 계산)
- 문제 3: 위의 통계량에 대한 p-값을 구하고 (반올림하여 소숫점 넷째 자리까지 계산), 유의수준 0.05하에서 가설검정의 결과를 채택/기각 중 하나로 선택하시오.
      

In [2]:
from google.colab import drive
drive.mount("/content/drive")

Mounted at /content/drive


## 데이터 불러오기

In [4]:
import pandas as pd
import numpy as np
import scipy.stats as stats

data = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/iM_alphaco/data_set/toothgrowth.csv')
data.head()

len supp  dose
0   4.2   VC   0.5
1  11.5   VC   0.5
2   7.3   VC   0.5
3   5.8   VC   0.5
4   6.4   VC   0.5

## 가설검정
- 독립표본 T-검정은 정규성 검정 뿐만 아니라, 두 그룹이 등분산성을 띠는지 확인하는 작업 필요
- Levene 검정을 사용한 등분산성 검정이 선행되어야 함

In [5]:
# prompt: Levene검정을 사용한 data의 등분산성 검정

oj_data = data[data['supp'] == 'OJ']['len']
vc_data = data[data['supp'] == 'VC']['len']

# Levene 검정 수행
stats.levene(oj_data, vc_data)

'''
LeveneResult(statistic=1.2135720656945064, pvalue=0.2751764616144053)
따라서 두 표본은 등분산성을 갖는다. (두 표본의 분산은 같다)
'''

LeveneResult(statistic=1.2135720656945064, pvalue=0.2751764616144053)

- p_value가 0.275 이므로 귀무가설을 채택한다.

## 문제 1
- 표본평균 $\overline{X_{OJ}}$, $\overline{X_{VC}}$를 구하시오.

In [9]:
oj_barx = oj_data.mean().round(2)
oj_barx

20.66

In [10]:
vc_barx = vc_data.mean().round(2)
vc_barx

16.96

## 문제 2
- 위의 가설을 검정하기 위한 검정통계량을 구하시오. (반올림하여 소숫점 둘째 자리까지 계산)

In [16]:
t_statistic, p_value = stats.ttest_ind(oj_data, vc_data)
t_statistic = round(t_statistic, 2)
p_value= round(p_value, 4)

t_statistic

1.92

## 문제 3.
- 위의 통계량에 대한 p-값을 구하고 (반올림하여 소숫점 넷째 자리까지 계산), 유의수준 0.05하에서 가설검정의 결과를 채택/기각 중 하나로 선택하시오.

In [17]:
p_value

0.0604

In [18]:
p_value < 0.05
#p값이 0.05보다 크기 때문에 이 경우 귀무가설을 채택한다.

False